warehouse/integrations/postgres/postgres.go

```diff
@@ -47,7 +47,7 @@ var errorsMappings = []model.JobError{
 		Format: regexp.MustCompile(`dial tcp: lookup .*: no such host`),
 	},
 	{
-		Type:   model.ResourceNotFoundError,
+		Type:   model.PermissionError,
 		Format: regexp.MustCompile(`dial tcp .*: connect: connection timed out`),
```

Comment:
The timeout could be for various reasons right?

Reply:
Yes.

Reply:
I was thinking of this database's firewall being one of the reasons.

Reply:
Also, since we don't have set connection limit on our end, it must be causing due to the connection limit set on the database side right.

Reply:
It could also be due to other network conditions as well, misconfiguration can be the most common. If after backoff retries we still get it, then the probabilities of miss-configuration are high (permission or incorrect DSN).

Reply:
Let me remove it.